cmake/configure/configure_symengine.cmake

```diff
@@ -17,11 +17,32 @@
 # Configuration for the SymEngine library:
 #
 
-#
-# We require at least version 0.3 of the symengine library:
-#
-SET(SYMENGINE_MINIMUM_REQUIRED_VERSION "0.3")
 
+MACRO(FEATURE_SYMENGINE_FIND_EXTERNAL var)
+  FIND_PACKAGE(SYMENGINE)
+
+  IF(SYMENGINE_FOUND)
+    SET(${var} TRUE)
+
+    #
+    # We require at least version 0.4 of the symengine library:
+    #
+    SET(_version_required "0.4")
```

Comment:
This should be given to `FIND_PACKAGE` call as CMake will choose an installation with version >=0.4 if there are multiple installations found. Without it CMake will choose one and it might be v0.3 even though there is another v0.4 installation.

Reply:
I prefer a descriptive error message and having to provide a better hint in case an insufficient version was found.

Reply:
With rare exceptions we use this logic with almost every other external dependency as well.